BluetoothLEExplorer/BluetoothLEExplorer/Models/ObservableBluetoothLEDevice.cs

```diff
@@ -570,6 +607,10 @@ public async Task<bool> Connect()
                     // Debugger break here so we can catch unknown exceptions
                     Debugger.Break();
                 }
+                finally
+                {
+                    bluetoothLEDeviceLock.Release();

```

Comment:
Consider 'using' statement for lock objects that can release when disposed (or file an issue to do it in another iteration).

Reply:
I filed an issue to figure out a better way of handling locks:  https://github.com/microsoft/BluetoothLEExplorer/issues/51